src/listing.rs

```diff
@@ -285,14 +284,26 @@ pub fn directory_listing<S>(
             .chunked()
             .body(Body::Streaming(Box::new(rx))))
     } else {
+        // Redirect to directory
+        if !renderer::has_trailing(&serve_path) {
+            let query = match req.query_string() {
+                "" => String::new(),
+                _ => format!("?{}", req.query_string())
+            };
+            return Ok(
+                HttpResponse::MovedPermanenty()
+                    .header("Location", format!("{}/{}", serve_path, query))
```

Comment:
Let's make it temporary. With Moved Permanently, you give browsers strong caching guarantees but what if the file system changes?

Reply:
The file pointed to by '/src' '/src/' is consistent, and ‘/src/’ will respond correctly even if the file changes.

Reply:
But what if `/src` becomes a file instead of a dir?

Reply:
This may cause the url to increase by ‘/’ but will not affect normal use
I try other web servers like nginx, they are the same

```
Client -> /src -> rewrite -> /src/ (dir) -> Ok 
```

```bash
mv src out && echo 123 > src
```

No cache
```
Client -> /src -> /src (file) -> Ok
```

Cache
```
Client -> /src -> rewrite -> /src/ (file) -> Ok
```
`miniserve` always responds correctly